File: app/core/backend-crypto-tracker/services/eth/etherscan-api.js
# app/services/eth/etherscan_api.py
import requests
import os

ETHERSCAN_API_KEY = os.getenv("ETHERSCAN_API_KEY")

def fetch_transaction(hash: str):
    url = "https://api.etherscan.io/api "
    params = {
        "module": "proxy",
        "action": "eth_getTransactionByHash",
        "txhash": hash,
        "apikey": ETHERSCAN_API_KEY
    }
    response = requests.get(url, params=params)
    return response.json().get("result")

File: app/core/backend-crypto-tracker/services/eth/etherscan-api.py
def get_eth_transaction(tx_hash):
    url = BASE_URL.format(tx_hash=tx_hash)
    response = requests.get(url)
    if response.status_code == 200:
        data = response.json()
        # Beispiel: Sender, Empfänger, Wert extrahieren
        return {
            "from": data["result"]["from"],
            "to": data["result"]["to"],
            "value": int(data["result"]["value"], 16) / 1e18  # Wei in ETH
        }
    else:
        raise Exception(f"ETH-API Fehler: {response.status_code}")
